🚗 Improved TeslaFi Sentry Mode Handling & Optimized Music Sync Process

jjermany opened the pull request:
🔹 Summary:

Improved TeslaFi-based Sentry Mode handling to ensure proper activation and deactivation.
Optimized Music Sync process to only notify users when actual file changes occur.
Improved logging consistency to match TeslaCam's archiving process.
🔹 Key Changes:
✅ Sentry Mode now correctly enables/disables only when necessary, preventing redundant API calls.
✅ TeslaFi wake/sleep commands removed, since Sentry Mode already keeps the car awake.
✅ Music Sync process now logs skipped/copied/deleted files properly, similar to TeslaCam.
✅ Edge cases handled: If TeslaCam doesn’t run but Music Sync does, Sentry Mode still turns off if it was originally OFF.
✅ Notifications are only sent when a sync or archive actually happens (no spam).

💡 Why this is useful:

Improves efficiency and reduces unnecessary TeslaFi API calls.
Provides better visibility into TeslaCam and Music Sync behavior.
Ensures users can easily track whether files were archived or skipped.
⚡ This PR is fully tested and ensures TeslaFi-based Sentry Mode behavior works correctly across all scenarios.

jjermany committed:
Removed redundancy and cleaned up sytax errors.

jjermany committed:
Merge branch 'marcone:main-dev' into main-dev

jjermany committed:
Update awake_start

Now we use TeslaFI telemetry data to gather the Sentry state. This code will ensure that we only turn on sentry if not already on and not turn off if sentry was already enabled.

jjermany committed:
Update awake_stop

jjermany committed:
Added sentry to our music sync process. Also enhanced logging and added some beneficial notifications

miles267 commented:
Hi @jjermany sorry dumb question but how do I take advantage of this? for example do I enable this: export BRANCH=jjermany and re-run the teslausb-upgrade script and it'll automatically update?  OR just wait for @marcone to merge into the default main-dev?

jjermany commented:
@miles267 Correct, you will need to change export BRANCH=jjermany and then run the upgrade script. Should work like a charm. Let me know if you have any issues here and ill help you out when I can.

miles267 commented:
Perhaps it's normal behavior but when I ran bin/setup-teslausb upgrade after adding export BRANCH=jjermany, it ran "silently." In that it did not show the line-by-line updates that it usually displays when it's using the default main-dev branch.  If that makes sense? I just wait and then the cursor re-appears at the ssh command prompt.  Which I assume means it ran without error.

jjermany commented:
@miles267 check and see if the awake_start, awake_stop files are updated accordingly. It should be logging when you run the upgrade command. It does for me when I use > export REPO=jjermany

jjermany commented:
@marcone @miles267 I am also working on a script that will automatically delete archived footage after a certain amount of days. The days set will be user defined based on variables in the config file. This is especially important for people who like to archive RecentClips. But there will be separate variables for each folder. If set to 0 than it will not delete anything for that folder.

miles267 commented:
@jjermany Great! I’ve since replaced export BRANCH=jermany with export REPO=jermany and it seems to have updated.

It is syncing though I now see in the archiveloop log that it's not indicating that it's starting and stopping sentry mode.  The way it was previously when I was using your awake_start and awake_stop modified files.  What entries are required in the .conf file now to ensure this occurs?

I currently have both of these (2) entries in my .conf:
export TESLA_WAKE_MODE=sentry
export TESLAFI_WAKE_MODE=sentry

jjermany commented:
@miles267 You do need to set the branch config since it will use main-dev by default. Make sure that is uncommented.

jjermany commented:
> @jjermany Great! I’ve since replaced export BRANCH=jermany with export REPO=jermany and it seems to have updated.
> 
> It is syncing though I now see in the archiveloop log that it's not indicating that it's starting and stopping sentry mode. The way it was previously when I was using your awake_start and awake_stop modified files. What entries are required in the .conf file now to ensure this occurs?
> 
> I currently have both of these (2) entries in my .conf: export TESLA_WAKE_MODE=sentry export TESLAFI_WAKE_MODE=sentry

With the new setup, you do not need to set export TESLA_WAKE_MODE=sentry or export TESLAFI_WAKE_MODE=sentry modes. It uses sentry mode by default now since its the only way to reliably keep the car awake anyway.

miles267 commented:
> With the new setup, you do not need to set export TESLA_WAKE_MODE=sentry or export TESLAFI_WAKE_MODE=sentry modes. It uses sentry mode by default now since its the only way to reliably keep the car awake anyway.

Thanks!  Now I've removed the TESLA_WAKE_MODE and TESLAFI_WAKE_MODE entries and have just:

export REPO=jjermany
export BRANCH=jjermany

And am re-running the upgrade script to commit the changes.  My DNS service is having issues today so it's been up and down (NextDNS).

jjermany commented:
> > With the new setup, you do not need to set export TESLA_WAKE_MODE=sentry or export TESLAFI_WAKE_MODE=sentry modes. It uses sentry mode by default now since its the only way to reliably keep the car awake anyway.
> 
> Thanks! Now I've removed the TESLA_WAKE_MODE and TESLAFI_WAKE_MODE entries and have just:
> 
> export REPO=jjermany export BRANCH=jjermany
> 
> And am re-running the upgrade script to commit the changes. My DNS service is having issues today so it's been up and down (NextDNS).

You will need to comment out the export BRANCH=jjermany. It uses the default main-dev so need to set this. Set this to # export BRANCH=jjermany

miles267 commented:
@jjermany am not sure it's working.  I don't see SENTRY being turned on and off as expected.  Here are the relevant entries in my .conf file.  Then re-ran upgrade and not seeing sentry toggled on (start) and off (end) or indications in the archiveloop.log.

'# export TESLA_WAKE_MODE=sentry
'# export TESLAFI_WAKE_MODE=sentry
'# export BRANCH=jjermany
export REPO=jjermany

jjermany commented:
When you ran the upgrade command did it log upgrade process like normal? If it didn't, try using export REPO=jjermany/teslausb instead. When you run upgrade and you dont see any logs it would typically indicate that the process is failing so its likely not grabbing repo. Hence the reason, you are not seeing any sentry logs. Its likely falling back to the marcone repo.

miles267 commented:
OK the upgrade runs with: export REPO=jjermany (it does not run with REPO=jjermany/teslausb

Regardless, upon updating after defining your REPO, there are no references to SENTRY in the archiveloop log (after upgrade + reboot) and observing the car itself, the SENTRY mode isn't being turned on/off.

jjermany commented:
Run the following command and send me the result,
`cat /root/.git/config | grep url`

If you get the following response, `-bash: git: command not found` then run this command AFTER you use bin/remountfs_rw command

`sudo apt update`
`sudo apt install -y git`

miles267 commented:
I've since run:
sudo apt update (successfully)
sudo apt install -y git (successfully)

However when I attempt:
cat /root/.git/config | grep url

It returns: No such file or directory.  I can confirm that since /root/.git or /root/git don't exist let alone /root/.git/config.

I can execute: git (as a command now) but it's not creating that directory structure.

I'm certainly glad to continue testing/troubleshooting as I'm excited to leverage this functionality.  But it doesn't quite seem production ready yet?

jjermany commented:
Try this command `git -C /root/teslausb remote -v`

jjermany commented:
> I've since run: sudo apt update (successfully) sudo apt install -y git (successfully)
> 
> However when I attempt: cat /root/.git/config | grep url
> 
> It returns: No such file or directory. I can confirm that since /root/.git or /root/git don't exist let alone /root/.git/config.
> 
> I can execute: git (as a command now) but it's not creating that directory structure.
> 
> I'm certainly glad to continue testing/troubleshooting as I'm excited to leverage this functionality. But it doesn't quite seem production ready yet?

What makes it even more difficult, is that I do not have the car at the moment. My gf has it since I work from home. It worked great for me last night so Im not sure why its not working. To me it seems like even though you are using jjermany for the repo its still not processing the correct awake_start and awake_stop scripts. nano into awake_start script and let me know if you see logging for the sentry mode.

miles267 commented:
OK.  So there isn't a path: /root/teslausb.  But there is a /teslausb.

git -C /root/teslausb remote -v naturally fails with: fatal: cannot change
to '/root/teslausb': No such file or directory

but this command: git -C /teslausb remote -v returns the following:

fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
***@***.***:/#

On Fri, Feb 28, 2025 at 1:14 PM jjermany ***@***.***> wrote:

> Try this command git -C /root/teslausb remote -v
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691373533>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKVGTFIGR62TCMJEG2UXPD2SCYQBAVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGM3TGNJTGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: jjermany]*jjermany* left a comment (marcone/teslausb#937)
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691373533>
>
> Try this command git -C /root/teslausb remote -v
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691373533>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKVGTFIGR62TCMJEG2UXPD2SCYQBAVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGM3TGNJTGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jjermany commented:
Okay this should fix it. Run this..

rm -rf /root/teslausb
git clone https://github.com/jjermany/teslausb.git /root/teslausb
cd /root/teslausb
git checkout main-dev

Then run 
git -C /root/teslausb remote -v

If the result shows my repo then run this,
bin/setup-teslausb upgrade

miles267 commented:
OK here is the result:


***@***.***:~# rm -rf /root/teslausb
***@***.***:~# git clone https://github.com/jjermany/teslausb.git
/root/teslausb
Cloning into '/root/teslausb'...
remote: Enumerating objects: 6702, done.
remote: Counting objects: 100% (1702/1702), done.
remote: Compressing objects: 100% (557/557), done.
remote: Total 6702 (delta 1281), reused 1152 (delta 1145), pack-reused 5000
(from 3)
Receiving objects: 100% (6702/6702), 2.77 MiB | 5.75 MiB/s, done.
Resolving deltas: 100% (4137/4137), done.
***@***.***:~# cd /root/teslausb
***@***.***:~/teslausb# git checkout main-dev
Already on 'main-dev'
Your branch is up to date with 'origin/main-dev'.
***@***.***:~/teslausb#

/root/teslausb folder now exists.  I then ran the teslausb upgrade script
again.  Despite that, no Sentry mode actions occurring.

***@***.***:~/teslausb# git -C /root/teslausb remote -v
origin  https://github.com/jjermany/teslausb.git (fetch)
origin  https://github.com/jjermany/teslausb.git (push)
***@***.***:~/teslausb#


On Fri, Feb 28, 2025 at 1:28 PM jjermany ***@***.***> wrote:

> Okay this should fix it. Run this..
>
> rm -rf /root/teslausb
> git clone https://github.com/jjermany/teslausb.git /root/teslausb
> cd /root/teslausb
> git checkout main-dev
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691395982>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKVGTEVR74MIKXGEZCI3AT2SC2ERAVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGM4TKOJYGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: jjermany]*jjermany* left a comment (marcone/teslausb#937)
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691395982>
>
> Okay this should fix it. Run this..
>
> rm -rf /root/teslausb
> git clone https://github.com/jjermany/teslausb.git /root/teslausb
> cd /root/teslausb
> git checkout main-dev
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691395982>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKVGTEVR74MIKXGEZCI3AT2SC2ERAVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGM4TKOJYGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jjermany commented:
Okay let me do some more testing when my car comes home after 5pm CST. Ill come back here and update you.

jjermany commented:
I think I may have figured out the issue. You need to enable TeslaFI's new telementry data logging on their website. This is how we get the current sentry state.  

login to your teslafi account > under settings, select "tesla account". In the meantime ill add some debugging to alert the user if this happens in the future.

miles267 commented:
That is definitely required though I’ve already had it enabled. In this case, something is preventing Teslausb from pulling down your scripts from the git repository. So it’s reverting to the default ones.
________________________________
From: jjermany ***@***.***>
Sent: Friday, February 28, 2025 2:47:27 PM
To: marcone/teslausb ***@***.***>
Cc: miles267 ***@***.***>; Mention ***@***.***>
Subject: Re: [marcone/teslausb] 🚗 Improved TeslaFi Sentry Mode Handling & Optimized Music Sync Process (PR #937)


I think I may have figured out the issue. You need to enable TeslaFI's new telementry data logging on their website. This is how we get the current sentry state.

login to your teslafi account > under settings, select "tesla account". In the meantime ill add some debugging to alert the user if this happens in the future.

—
Reply to this email directly, view it on GitHub<https://github.com/marcone/teslausb/pull/937#issuecomment-2691518907>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANKVGTDJDRSCHYYE7OQ5SRD2SDDN7AVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGUYTQOJQG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[jjermany]jjermany left a comment (marcone/teslausb#937)<https://github.com/marcone/teslausb/pull/937#issuecomment-2691518907>

I think I may have figured out the issue. You need to enable TeslaFI's new telementry data logging on their website. This is how we get the current sentry state.

login to your teslafi account > under settings, select "tesla account". In the meantime ill add some debugging to alert the user if this happens in the future.

—
Reply to this email directly, view it on GitHub<https://github.com/marcone/teslausb/pull/937#issuecomment-2691518907>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANKVGTDJDRSCHYYE7OQ5SRD2SDDN7AVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGUYTQOJQG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

jjermany committed:
Added debug for cases where the user does not have fleet telementry enabled.

jjermany commented:
Nope, the fleet telemetry data that I am referring to is different than the commands used for controlling sentry. You have to manually enable it as its not on by default. If you already have it on then yes. I added the debugging so if you run the upgrade script and watch the logs it should now show something at the very least. If its not than then yes something is wrong with the way that the project is cloning repositories. 
![image](https://github.com/user-attachments/assets/2a4a9c5d-c881-4099-9b59-26adf3a15ca0)

miles267 commented:
Yes. Sorry, I already had fleet telemetry enabled in TeslaFi. I was using your previous modified awake_start and awake_stop scripts which required that be enabled also for Teslausb. However now it appears Teslausb is reverting back to those default scripts from the main-dev and not acquiring yours. Not yet sure how to get that part of this working. Happy to attach any files and or additional command results as needed here. Thanks.

jjermany commented:
If you nano into the awake_start script does it reflect what is currently in my repo? setup logs or diagnostics may also give us a clue. If you scan those logs and just see if you see any mention of my repo anywhere. 

If all else fails, you can just nano and edit the awake_start and awake_stop scripts manually. Ill do more when the car arrives at home. I want to see if it will continue to log correctly for me. I am pretty sure I checked all the boxes in my testing last night but Ill know for sure when the car starts its sync process later.

miles267 commented:
Yes, when I review awake_start it does appear to reference your TeslaFi
routines.

On Fri, Feb 28, 2025 at 3:28 PM jjermany ***@***.***> wrote:

> If you nano into the awake_start script does it reflect what is currently
> in my repo? setup logs or diagnostics may also give us a clue. If you scan
> those logs and just see if you see any mention of my repo anywhere.
>
> If all else fails, you can just nano and edit the awake_start and
> awake_stop scripts manually. Ill do more when the car arrives at home. I
> want to see if it will continue to log correctly for me. I am pretty all
> the boxes were checked in my testing last night but Ill know for sure when
> the car starts its sync process later.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691575955>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKVGTDPGHP6ZBGTZZC2ZYT2SDIJNAVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGU3TKOJVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: jjermany]*jjermany* left a comment (marcone/teslausb#937)
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691575955>
>
> If you nano into the awake_start script does it reflect what is currently
> in my repo? setup logs or diagnostics may also give us a clue. If you scan
> those logs and just see if you see any mention of my repo anywhere.
>
> If all else fails, you can just nano and edit the awake_start and
> awake_stop scripts manually. Ill do more when the car arrives at home. I
> want to see if it will continue to log correctly for me. I am pretty all
> the boxes were checked in my testing last night but Ill know for sure when
> the car starts its sync process later.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcone/teslausb/pull/937#issuecomment-2691575955>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKVGTDPGHP6ZBGTZZC2ZYT2SDIJNAVCNFSM6AAAAABYCQ3YUKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGU3TKOJVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jjermany commented:
Okay that tells me that is indeed using my scripts. I added some debug for cases where teslafi telementry is not enabled. Just so I feel better about this, can you run the upgrade and try again. Let me know if you see anything about teslafi.

miles267 commented:
Upgraded by still not working.  BTW here's TeslaFi confirmation of telemetry enabled.  And I can send commands to car (including Sentry) from TeslaFi without issue.
![Screenshot 2025-02-28 163632](https://github.com/user-attachments/assets/d78ddaf1-353c-4a23-868a-2ed77064c7ef)

jjermany commented:
Thank you for confirming. I figured out the issue. My script is using jq and this command is not installed by default by the TeslaUSB setup. Im going to find another way to write so that it does not rely on that command. Once I installed the jq, the logs showed up. Ill let you know when I have it fixed.

jjermany committed:
Removed jq command when trying to get sentry status.

jjermany committed:
Allows you to connect external wireless adapters

jjermany committed:
Update configure-ap.sh

Changing back to the original. My version failed. May be forced to try a different network adapter or not use one at all.

jjermany commented:
@miles267 Okay its working. Once you run the upgrade you should now see the logs.

miles267 commented:
> @miles267 Okay its working. Once you run the upgrade you should now see the logs.

WONDERFUL!  I just did nothing more than run the teslausb upgrade script (based on the previous changes we already made) and it's now working as you've indicated.

Q - if I now add the REPO line to another car and run the upgrade script, is that all that is required based on your code changes (I have another car to update now)? I cannot rem how much of what I did above was needed before the upgrade addressed the critical parts.

jjermany commented:
Just what I like to hear! Yes, all you have to do now is just add the repo to the config before you install and it will use it. No config changes are necessary.

jjermany committed:
Update awake_stop

Added push notifications for when Sentry mode gets turned off.

miles267 commented:
@jjermany I'm using this now (since all of your updates), however it doesn't appear as if the awake stop is turning off Sentry mode via TeslaFi at the end of the sync.  It does reliably start/enable Sentry Mode though.  I saw that you recently pushed an awake_stop update which I've applied, but it still doesn't seem to be working?

jjermany commented:
@miles267 its turning off for me using the same updates. 

Can you force an archive using web UI or command and send me the log it generates for that specific archive process? 

One thing I do know I need to change is that it turns sentry mode on and off even if there is nothing to archive. So it will turn it on then back off. I want to change it to where it only uses sentry commands where absolutely necessary.

miles267 commented:
> @miles267 its turning off for me using the same updates.
> 
> Can you force an archive using web UI or command and send me the log it generates for that specific archive process?
> 
> One thing I do know I need to change is that it turns sentry mode on and off even if there is nothing to archive. So it will turn it on then back off. I want to change it to where it only uses sentry commands where absolutely necessary.

Apologies.  I just re-upgraded on both of our cars with your REPO set and it's working as I expected now.  False alarm!

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update archiveloop

Fixed delay issue.

jjermany committed:
Update configure-ap.sh

lets you use external adapters during upgrade now.

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

Going back to default until I have fix.

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh

jjermany committed:
Update configure-ap.sh
